### PR TITLE
ZEPPELIN-1577. LivyInterpreter should not use FIFOScheduler

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?

This PR create a new scheduler FIFOPerUserScheduler. So that each user's request is isolated. and it is FIFOSchduler per user. 
### What type of PR is it?

[Bug Fix | Improvement]
### Todos
- [ ] - Task
### What is the Jira issue?
- https://issues.apache.org/jira/browse/ZEPPELIN-1577
### How should this be tested?

Tested manually 
### Questions:
- Does the licenses files need update? No
- Is there breaking changes for older versions? No
- Does this needs documentation? No

### CODE REVIEW

**Comment by zjffdu** [comment]
Will add test after #1462

**Comment by zjffdu** [comment]
@prabhjyotsingh @felixcheung  Please help review.

**Comment by felixcheung** [comment]
Probably something @Leemoonsoo should review

File: zeppelin-interpreter/src/main/java/org/apache/zeppelin/scheduler/FIFOPerUserScheduler.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+ */
+public class FIFOPerUserScheduler implements Scheduler {
+
+  private static Logger LOGGER = LoggerFactory.getLogger(FIFOScheduler.class);
+
+  private ExecutorService executor;
+  private SchedulerListener listener;
+  boolean terminate = false;
+  private String name;
+  private int maxUser = 10;
```
Comment (Leemoonsoo) [inline zeppelin-interpreter/src/main/java/org/apache/zeppelin/scheduler/FIFOPerUserScheduler.java]: Looks like unused var. Is it?

**Comment by bikassaha** [comment]
Perhaps for this issue and other similar issues related to multiple users, the mode for the livy interpreter should be scoped instead of shared. If we run 1 interpreter per user then that matches the end to end flow wrt sessions. Then interpreters will have their own queue and not need this workaround. Also one user restarting their interpreter would not affect existing interpreter instances for others users. Is that a correct understanding?
Reply

**Comment by Leemoonsoo** [comment]
@bikassaha Good point. I Agree, that approach seems better.

**Comment by zjffdu** [comment]
Yes, that would be better. I think currently zeppelin hard coded the default mode (shared mode), there's no way to set the default mode for interpreter, we may add one option in interpreter-setting.json.  And also we may add one option to restrict the mode usage as for livy only scoped is valid.

**Comment by bikassaha** [comment]
@zjffdu Isolated would also be valid but perhaps an overkill. So default of scoped makes sense.

**Comment by Leemoonsoo** [comment]
Exactly. Adding option in interpreter-setting.json and restrict mode usage is great idea. Similar approaches has been made in ZEPPELIN-1026 for editor mode restriction.

**Comment by zjffdu** [comment]
Close this one, and open #1585 to allow user to set default mode in interpreter-setting.json

**Pull request closed without merging**